src/Raven.Server/ServerWide/RawDatabaseRecord.cs

```diff
@@ -972,43 +972,38 @@ public Dictionary<string, AiAgentConfiguration> AiAgents
 
                 if (_aiAgents == null)
                 {
-                    _aiAgents = [];
-                    if (_record.TryGet(nameof(DatabaseRecord.AiAgents), out BlittableJsonReaderObject agents) && agents != null)
+                    _aiAgents = new ();
+                    if (_record.TryGet(nameof(DatabaseRecord.AiAgents), out BlittableJsonReaderArray bjra) && bjra != null)
                     {
-                        var propertyDetails = new BlittableJsonReaderObject.PropertyDetails();
-                        for (var i = 0; i < agents.Count; i++)
-                        {
-                            agents.GetPropertyByIndex(i, ref propertyDetails);
-
-                            if (propertyDetails.Value == null)
-                                continue;
-
-                            if (propertyDetails.Value is BlittableJsonReaderObject agent)
-                                _aiAgents[propertyDetails.Name] = JsonDeserializationCluster.AiAgentConfiguration(agent);
-                        }
+                        foreach (BlittableJsonReaderObject element in bjra)
+                            _aiAgents.Add(JsonDeserializationCluster.AiAgentConfiguration(element));
                     }
                 }
 
                 return _aiAgents;
             }
         }
 
-        public bool TryGetAiAgent(string name, out AiAgentConfiguration config)
+        public bool TryGetAiAgent(string identifier, out AiAgentConfiguration config)
         {
+            const string identifierName = nameof(AiAgentConfiguration.Identifier);
+
             config = null;
             if (_materializedRecord != null)
             {
-                config = _materializedRecord.AiAgents[name];
+                config = _materializedRecord.AiAgents.FirstOrDefault(c => c.Identifier == identifier);
```

Comment:
you can return `true` if the agent doesn't exists?

Reply:
case sensitivity

Reply:
https://github.com/ravendb/ravendb/pull/20906#discussion_r2191759806